src/mbgl/style/expression/is_constant.cpp

```diff
@@ -49,6 +49,19 @@ bool isZoomConstant(const Expression& e) {
     return isGlobalPropertyConstant(e, std::array<std::string, 1>{{"zoom"}});
 }
 
+bool isRuntimeConstant(const Expression& expression) {
+    if (expression.getKind() == Kind::ImageExpression) {
+        return false;
+    }
+
+    bool runtimeConstant = true;
+    expression.eachChild([&](const Expression& e) {
+        if (runtimeConstant && !isRuntimeConstant(e)) {
+            runtimeConstant = false;
```

Comment:
return false ?

Reply:
function returns void, setting flag to false is the only way to short-circuit recursion

Reply:
:crying_cat_face: